### PR TITLE
[DO Not Merge] VTGATE SPLIT QUERY endtoend test cases in GO

### PR DESCRIPTION
**Pull request by ajeetj**
VTGATE SPLIT QUERY endtoend test cases in GO using the new cluster.

@deepthi
Question: Split query hangs in the terminal, we try to exec command in another window but got the same result. What can we do to solve this?

Hangs at `vtctlclient -server localhost:27663 VtGateSplitQuery -keyspace ks -split_count 2 "select id, name from user"`

Reference image
Screen Shot 2019-10-22 at 11 52 24 AM

### CODE REVIEW

**Comment by deepthi** [comment]
Is this working now or still hanging?

**Comment by ajeetj** [comment]
> Is this working now or still hanging?

This one still hangs.
In my other test case (that I am currently working) I use vttablet commands via vtctlclients and they all work fine.

File: go/test/endtoend/cluster/vtctlclient_process.go [hunk 1]
```diff
@@ -96,3 +96,18 @@ func VtctlClientProcessInstance(hostname string, grpcPort int, tmpDirectory stri
 	}
 	return vtctlclient
 }
+
+// VtGateSplitQuery applies vitess schema (JSON format) to the keyspace
+func (vtctlclient *VtctlClientProcess) VtGateSplitQuery(keyspace string, sql string, splitCount int) (string, error) {
+	tmpProcess := exec.Command(
+		vtctlclient.Binary,
+		"-server", vtctlclient.Server,
```
Comment (deepthi) [inline go/test/endtoend/cluster/vtctlclient_process.go:104]: Looks like this is being run against vtctld, it needs to be run against vtgate.
When I tried running locally, the command being run is:
```
vtctlclient -server localhost:22400 VtGateSplitQuery -keyspace ks -split_count 2 "select id, name from user"
```
22400 is the vtctld port number
```
vtctld.INFO:I1101 14:50:44.985899    9084 grpc_server.go:212] Listening for gRPC calls on port 22400
```
vtgate port number is 22420
```
vtgate.INFO:I1101 14:51:12.387798   13823 grpc_server.go:212] Listening for gRPC calls on port 22420
```

Comment (deepthi) [reply go/test/endtoend/cluster/vtctlclient_process.go:104]: The larger question is about the functionality itself. According to comments in the original python test, this is for MapReduce functionality.
Is that something we are keeping? Is it v2 or v3?
@sougou

Comment (sougou) [reply go/test/endtoend/cluster/vtctlclient_process.go:104]: You don't have to port SplitQuery. I plan to delete it when I start the v2 cleanup.

**Comment by deepthi** [comment]
@arindamnayak this also means that you can delete the SplitQuery portion of initial_sharding_test

File: go/test/endtoend/cluster/vtctlclient_process.go (hunk 1 above)
Comment (deepthi) [reply go/test/endtoend/cluster/vtctlclient_process.go:104]: @arindamnayak this also means that you can delete the SplitQuery portion of initial_sharding_test

Comment (ajeetj) [reply go/test/endtoend/cluster/vtctlclient_process.go:104]: Great, I will close this PR then. 
I was about to post the question "What is the use of vtgate SplitQuery" in the channel.

**Comment by ajeetj** [comment]
NOT required, closing this PR

**Pull request closed without merging**